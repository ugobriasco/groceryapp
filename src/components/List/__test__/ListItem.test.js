import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

import {ListItem, styles} from '../';

describe('ListItem', () => {

	it('renders correctly', () => {
	const tree = renderer.create(<ListItem />)

	});

	it ('styles an object', () => {
		expect(typeof styles).toBe('object');
	});

	describe('text and custom icon props', () => {
		it('should render a text and a custom element if passed ', () => {
			const rendered = renderer.create(<ListItem text="foo" customIcon={<Text>This is a custom element</Text>}/>);
			expect(rendered).toMatchSnapshot();	
		});
	});

	describe('onPress prop', () => {
		it('should trigger a function if pressed ', () => {
			const rendered = renderer.create(<ListItem onPress={()=>console.log('foo')}/>);
			expect(rendered).toMatchSnapshot();	
		});
	});

});



