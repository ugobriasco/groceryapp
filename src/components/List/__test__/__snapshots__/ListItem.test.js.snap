// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem onPress prop should trigger a function if pressed  1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    />
  </View>
</View>
`;

exports[`ListItem text and custom icon props should render a text and a custom element if passed  1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={undefined}
    >
      foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      This is a custom element
    </Text>
  </View>
</View>
`;
